src/types.rs

```diff
@@ -271,23 +279,79 @@ impl Type {
     pub(crate) fn validate(
```

Comment:
This function could use a docstring

Reply:
Yes, good point :). I've done `TypeArg::validate` as well as that's also `pub(crate)` but the other `fn validate`s are less visible so I haven't done so there.